graphman-client/modules/graphman.js

```diff
@@ -43,9 +44,23 @@ module.exports = {
             method: 'POST',
             rejectUnauthorized: gateway.rejectUnauthorized,
             headers: headers,
-            auth: gateway.username + ":" + gateway.password,
             body: body || {}
         };
+        
+        const isUsernamePasswordAuthentication = gateway.username && gateway.password
+        const isCertificateAuthentication = gateway.certificateKeyName && gateway.certificateCertName
+
+        if(!isUsernamePasswordAuthentication || !isCertificateAuthentication) {
+            throw new Error("Either username/password or certificate authentication must be configured. Please provide the following values: \n 'Username/Password Authentication: the username and password fields' \n 'Certificate Authentication: certificateKeyName and certificateCertName fields'.");
+        }
+        if (isUsernamePasswordAuthentication) {
+            req.auth = gateway.username + ":" + gateway.password;
+        }
+        if (isCertificateAuthentication) {
+            // This expects the certificate.pem and certificate.key file(s) to be in the graphman-client directory. 
+            req.key = fs.readFileSync(`${__dirname}/../${gateway.certificateKeyName}`);
+            req.cert = fs.readFileSync(`${__dirname}/../${gateway.certificateCertName}`);
+        }
```

Comment:
we can improve this code for more readability
```
if (gateway.username && ...) {
} else if (gateway.certFilename && ...) {
} else {
  throw new Error...
}
```

Reply:
Changed to your suggested code